### PR TITLE
Fix webView waits, workaround for android O

### PR DESCRIPTION
**Pull request by npark-mozilla**

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/focus/activity/ImageSelectTest.java [hunk 1]
```diff
@@ -133,7 +133,8 @@ public void ImageMenuTest() throws InterruptedException, UiObjectNotFoundExcepti
         TestHelper.inlineAutocompleteEditText.setText(webServer.url(TEST_PATH).toString());
         TestHelper.hint.waitForExists(waitingTime);
         TestHelper.pressEnterKey();
-        TestHelper.webView.waitForExists(waitingTime);
+        assertTrue(TestHelper.webView.waitForExists(waitingTime));
+        titleMsg.waitForExists(waitingTime);
```
Comment (pocmo) [inline app/src/androidTest/java/org/mozilla/focus/activity/ImageSelectTest.java]: Can we add a assertTrue here too?

File: app/src/androidTest/java/org/mozilla/focus/activity/SwitchContextTest.java [hunk 2]
```diff
@@ -187,7 +187,8 @@ public void settingsToFocus() throws InterruptedException, UiObjectNotFoundExcep
         TestHelper.pressEnterKey();
 
         // Assert website is loaded
-        TestHelper.webView.waitForExists(waitingTime);
+        assertTrue(TestHelper.webView.waitForExists(waitingTime));
+        titleMsg.waitForExists(waitingTime);
```
Comment (pocmo) [inline app/src/androidTest/java/org/mozilla/focus/activity/SwitchContextTest.java]: .. and here :)

File: app/src/androidTest/java/org/mozilla/focus/activity/helpers/HostScreencapScreenshotStrategy.java [hunk 3]
```diff
@@ -55,7 +55,7 @@ public void takeScreenshot(String screenshotName, ScreenshotCallback screenshotC
             bitmap = createDummyScreenShot();
         }
 
-        screenshotCallback.screenshotCaptured(screenshotName, bitmap);
+//        screenshotCallback.screenshotCaptured(screenshotName, bitmap);
```
Comment (pocmo) [inline app/src/androidTest/java/org/mozilla/focus/activity/helpers/HostScreencapScreenshotStrategy.java]: Um. No!

File: app/src/androidTest/java/org/mozilla/focus/activity/screenshots/ScreenGrabTest.java [hunk 4]
```diff
@@ -304,17 +305,27 @@ private void takeAddToHomeScreenScreenshot() throws UiObjectNotFoundException {
         assertTrue(TestHelper.AddtoHSCancelBtn.waitForExists(waitingTime));
         Screengrab.screenshot("AddtoHSDialog");
 
-        TestHelper.AddtoHSCancelBtn.click();
+        TestHelper.pressBackKey();
+        TestHelper.pressBackKey();
```
Comment (pocmo) [inline app/src/androidTest/java/org/mozilla/focus/activity/screenshots/ScreenGrabTest.java]: Why?

File: app/src/androidTest/java/org/mozilla/focus/activity/screenshots/ScreenGrabTest.java [hunk 5]
```diff
@@ -305,16 +306,26 @@ private void takeAddToHomeScreenScreenshot() throws UiObjectNotFoundException {
         Screengrab.screenshot("AddtoHSDialog");
 
         TestHelper.AddtoHSCancelBtn.click();
+        TestHelper.pressBackKey();
```
Comment (pocmo) [inline app/src/androidTest/java/org/mozilla/focus/activity/screenshots/ScreenGrabTest.java]: I still do not understand why we press back here now?

Comment (npark-mozilla) [reply app/src/androidTest/java/org/mozilla/focus/activity/screenshots/ScreenGrabTest.java]: you're right, I was thinking about keyboard dismissing, but i shouldn't do that if I was pressing cancel.

File: app/src/androidTest/java/org/mozilla/focus/activity/screenshots/ScreenGrabTest.java [hunk 6]
```diff
@@ -482,16 +476,21 @@ private void takeScreenshotOfErrorPages(UiDevice device) throws UiObjectNotFound
...
-                    .perform(webScrollIntoView());
-
+            // Android O has an issue with using Locator.ID
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                UiObject tryAgainBtn = device.findObject(new UiSelector()
+                        .descriptionContains(context.getString(R.string.errorpage_refresh))
+                        .clickable(true));
+                assertTrue(tryAgainBtn.waitForExists(waitingTime));
+            } else {
+                onWebView()
```
Comment (sputnikci) [inline app/src/androidTest/java/org/mozilla/focus/activity/screenshots/ScreenGrabTest.java]: \[Checkstyle\] ERROR: '\+' is not preceded with whitespace\.

Comment (npark-mozilla) [reply app/src/androidTest/java/org/mozilla/focus/activity/screenshots/ScreenGrabTest.java]: hmm, lint bug perhaps?

Comment (pocmo) [reply app/src/androidTest/java/org/mozilla/focus/activity/screenshots/ScreenGrabTest.java]: It just disappeared. :)

**Pull request merged**